drivers/trim_util.py

```diff
@@ -104,15 +104,26 @@ def do_trim(session, args):
         time.sleep(LOCK_RETRY_INTERVAL)
 
     if got_lock:
-        try:
-            vg_name = _vg_by_sr_uuid(sr_uuid)
-            lv_name = sr_uuid + TRIM_LV_TAG
-            lv_path = _lvpath_by_vg_lv_name(vg_name, lv_name)
-
-            # Clean trim LV in case the previous trim attemp failed
-            if lvutil.exists(lv_path):
-                lvutil.remove(lv_path)
+        vg_name = _vg_by_sr_uuid(sr_uuid)
+        lv_name = sr_uuid + TRIM_LV_TAG
+        lv_path = _lvpath_by_vg_lv_name(vg_name, lv_name)
+
+        # Clean trim LV in case the previous trim attemp failed
+        if lvutil.exists(lv_path):
+           lvutil.remove(lv_path)
+
+        #Check if VG limits are enough for creating LVM.
+        if lvutil._getVG_freespace(vg_name) < lvutil.LVM_SIZE_INCREMENT:
+           util.SMlog("No space to claim on a completely filled SR %s to perform lvutil.create" % sr_uuid)
+           err_msg = {ERROR_CODE_KEY: 'Trim failed on completely filled SR',
+                      ERROR_MSG_KEY: 'No space to claim on a completely filled SR'}
+           if lvutil.exists(lv_path):
```

Comment:
This block of code duplicates logic that is already happening further down in do_trim. Can't it be reused?